xtevent/_eventgenvars.ado

```diff
@@ -1,306 +1,515 @@
+					by `panelvar' : egen double `minz' = min(`z') if `touse'
+				}
+				* Exclude units with missing event-time
+				foreach x of varlist _k_eq* {
+					qui replace `x' = . if `etmismax'==1 & `mz'>0 & `minz'==0 & `touse'
+				}
+				qui replace __k = . if `etmismax'==1 & `touse'
+				qui levelsof `panelvar' if `etmismax'==1 & `mz'>0 & `minz'==0 & `touse', loc(mis)
+				foreach j in `mis' {
+					di as txt _n "Unit `j' not used because of ambiguous event-time due to missing values in policyvar."
+				}
+				qui replace `touse' = 0 if `etmismax'==1 & `mz'>0 & `minz'==0 & `touse'			
+			}
+		}	
+		
+			
+		* Set omitted variable.
+		unab included : _k*
+		loc toexc ""
+		foreach j in `norm' {
+			loc abs = abs(`j')
+			if `j'<0 loc toexc "`toexc' _k_eq_m`abs'"
+			else if `j'>=0 loc toexc "`toexc' _k_eq_p`abs'"
+		}
+		loc included: list local included - toexc
+		
+		* Group k
+		qui gen `kg' = __k if `touse'
+		* Group if outside window
+		qui replace `kg' = `=`lwindow'-1' if __k < `=`lwindow'' & `touse'
+		qui replace `kg' = `=`rwindow'+1' if __k >= `rwindow' & `touse'
+		qui levelsof `kg', loc(kgs)
+		
+		* If extrapolating a linear trend, exclude some of the event time dummies
+		loc komittrend ""
+		* di "`included'"
+		if "`trend'" != "" {
+			if `bin'!=1 {
+				di as err _n "Cannot extrapolate linear trend with non-binary policyvar."
+				exit 301
+			}
+		
+			* Generate the trend		
+			qui gen _ttrend = __k  if `touse'	
+			qui replace _ttrend = 0 if !inrange(_ttrend,`lwindow',`=`rwindow'') & `touse'
+			* qui replace _ttrend = 0 if _ttrend<`=`trend'' & `touse'
+			qui replace _ttrend = 0 if mi(_ttrend) & `touse'
+			la var _ttrend trend
+			
+			
+			* Exclude the coefficients that are restricted: Those on the negative values of the trend range, plus the endpoints
+			
+			loc komittrend ""	
+			
+			foreach klevel in `kgs' {
+				if (inrange(`klevel',`trend',-1) & `klevel' != -1) {
+					loc absk = abs(`klevel')
+					if `klevel'<0 loc plus = "m"
+					else loc plus "p"
+					loc l`plus'`absk' "`plus'`absk'"
+					loc toexc "_k_eq_`l`plus'`absk''"
+					loc included: list included - toexc
+					loc komittrend = "`komittrend' `klevel'"				
+				}
+			*noi di "***"
+			*noi di "`included'"
+			}
+			* loc toexc: word 1 of `included'
+			*loc included: list local included - toexc
+			*loc total: word count `included'
+			*loc toexc: word `total' of `included'
+			*loc included: list local included - toexc
+			* di "`included'"
+		}	
+		
+		* Save the names of included reg vars to extract these coefficients from the overall matrix later
+		
+		loc j=1
+		loc names ""
+		foreach var in `included' {
+			if `j'==1 loc names `""`var'""'
+			else loc names `"`names'.."`var'""'
+			loc ++ j
+		}	
+		return local included = "`included'"
+		return local names = `"`names'"'
+		return local komittrend= "`komittrend'"	
+		
+	}
+	******* add the imputed policyvar to the database 
+	
+	if "`impute'"!="" {
+		cap drop `policyvar'_imputed

```

Comment:
@Constantino-Carreto-Romero this `capture drop` is unnecessary

Reply:
Resolved in https://github.com/JMSLab/xtevent/pull/24/commits/36dd2c21e2cd33babfa1a1103cc3db99781a07d5